### PR TITLE
fix: toggle report missing settings

### PR DESCRIPTION
**Pull request by robrap**
**Description:**

Adds missing settings to the toggle report.

Fixes https://github.com/openedx/edx-toggles/issues/310

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -14,6 +14,11 @@ Change Log
 Unreleased
 ~~~~~~~~~~
 
+[5.1.1] - 2024-01-31
```
Comment (timmc-edx) [inline CHANGELOG.rst:17]: Remember to bump the version in code as well.

File: edx_toggles/toggles/state/internal/report.py [hunk 2]
```diff
@@ -213,7 +213,14 @@ def _add_settings(settings_dict):
     """
     Fill the `settings_dict`: will only include values that are set to true or false.
     """
-    for setting_name, setting_value in vars(settings).items():
```
Comment (timmc-edx) [inline edx_toggles/toggles/state/internal/report.py:216]: Was the issue that `vars` just doesn't work at all with the settings object?

**Review by timmc-edx — APPROVED** [review]

File: CHANGELOG.rst (hunk 1 above)
Comment (robrap) [reply CHANGELOG.rst:17]: Someone already updated the version here: https://github.com/openedx/edx-toggles/blob/c9ed5b02e39382266863abda35c95f81be3e2d39/edx_toggles/__init__.py#L5 
But it was never released: https://pypi.org/project/edx-toggles/

File: edx_toggles/toggles/state/internal/report.py (hunk 2 above)
Comment (robrap) [reply edx_toggles/toggles/state/internal/report.py:216]: It works, but only for settings that were previously requested. So if I set a new test setting, and then if I retrieve that setting in middleware, before the view that uses this report, then it starts appearing. I imagine this is due to lazy loading of settings.

Comment (robrap) [reply edx_toggles/toggles/state/internal/report.py:216]: Note: I did ensure that the new test I added failed without the fix.

File: CHANGELOG.rst (hunk 1 above)
Comment (timmc-edx) [reply CHANGELOG.rst:17]: Hah, OK. :-)

File: edx_toggles/toggles/state/internal/report.py (hunk 2 above)
Comment (timmc-edx) [reply edx_toggles/toggles/state/internal/report.py:216]: Ah, very interesting. And that makes sense.

**Pull request merged**